### DISCUSSION TITLE
How to dinstinguish between json object and json array

### DISCUSSION THREAD

**Discussion by JamesInform** [jackc/pgx#1305] [Q&A]
In PostgreSQL a json / jsonb field can store both, a json object as well as a json array.

What is the best solution to scan a row in pgx if you don't know ahead if a queried json is an object or an array?

This seems to only work for json array:
```
var js []map[string]interface{}
...
rows.Scan(&js)
```

This seems to only work for json objects:
```
var js map[string]interface{}
...
rows.Scan(&js)
```

Is there a universial solutions for handling json in pgx?

Cheers,
James

**Comment by jackc** [comment]
Internally, it is calling `json.Unmarshal()`. So most techniques you would use there should work with pgx.

In particular scan into `any`.

```go
var js any
...
rows.Scan(&js)
```
